src/demo/helpers.js

```diff
@@ -137,3 +144,28 @@ export const generateColorPalette = (colors, bodyIndex = null) => {
     knnData: bodyColor.knnData
   };
 };
+
+export const currentRunTime = state => {
+  let t = 0;
+  if (state.isRunning) {
+    if (!state.lastStartTime) {
+      state = setState({lastStartTime: $time()});
```

Comment:
Is it odd for a getter function to be setting state?

Reply:
it is! i'll move that into the main renderer method, since it's already doing some check-state-and-set-state-appropriately logic